lib/api_client/call_analysis_api.js

```diff
@@ -0,0 +1,38 @@
+const utils = require("../utils");
+const config = require("../config");
+const ensureOption = require('../utils/ensureOption').defaults(config());
+const execute_request = require("./execute_request");
+
+const {ensurePresenceOf} = utils;
+
+function call_analysis_api(method, uri, params, callback, options) {
+  ensurePresenceOf({
+    method,
+    uri
+  });
+  const api_url = utils.base_api_url_v2()(uri, options);
+  let auth = {};
+  if (options.oauth_token || config().oauth_token) {
+    auth = {
+      oauth_token: ensureOption(options, "oauth_token")
+    };
+  } else {
+    auth = {
+      key: ensureOption(options, "api_key"),
+      secret: ensureOption(options, "api_secret")
+    };
+  }
+  options.content_type = 'json';
+
+  return execute_request(method, params, auth, api_url, callback, options)
+    .then(requestResult => {
```

Comment:
What happens on error?
We can have errors that look like this
```
{
  "error": {
    "category": "auth_error",
    "code": "MA_00010",
    "message": "authentication error",
    "details": {
      "message": "api_secret mismatch"
    }
  }
}
```

or we might have something else (unstructured) returned if something went really wrong.

Reply:
It'll be returned correctly - I verified by sending incorrect `analysis_type` and credentials and got two different errors but looking exactly like you sent.